pkg/plugin/backup_item_action.go

```diff
@@ -38,41 +38,43 @@ import (
 // interface.
 type BackupItemActionPlugin struct {
 	plugin.NetRPCUnsupportedPlugin
-	impl arkbackup.ItemAction
-	log  *logrusAdapter
+
+	*serverMux
 }
 
 // NewBackupItemActionPlugin constructs a BackupItemActionPlugin.
-func NewBackupItemActionPlugin(itemAction arkbackup.ItemAction) *BackupItemActionPlugin {
+func NewBackupItemActionPlugin() *BackupItemActionPlugin {
 	return &BackupItemActionPlugin{
-		impl: itemAction,
+		serverMux: newServerMux(),
 	}
 }
 
-func (p *BackupItemActionPlugin) Kind() PluginKind {
-	return PluginKindBackupItemAction
-}
-
-// GRPCServer registers a BackupItemAction gRPC server.
-func (p *BackupItemActionPlugin) GRPCServer(s *grpc.Server) error {
-	proto.RegisterBackupItemActionServer(s, &BackupItemActionGRPCServer{impl: p.impl})
-	return nil
-}
+//////////////////////////////////////////////////////////////////////////////
+// client code
+//////////////////////////////////////////////////////////////////////////////
 
-// GRPCClient returns a BackupItemAction gRPC client.
+// GRPCClient returns a clientMux for BackupItemAction gRPC clients.
 func (p *BackupItemActionPlugin) GRPCClient(c *grpc.ClientConn) (interface{}, error) {
-	return &BackupItemActionGRPCClient{grpcClient: proto.NewBackupItemActionClient(c), log: p.log}, nil
+	return newClientMux(c, newBackupItemActionGRPCClient), nil
```

Comment:
what did you think about renaming `clientMux` -> `clientDispenser` (and this func as well)?

Reply:
Yes, we talked about that, but I overlooked it. I'll make the change.